external/CMakeLists.txt

```diff
@@ -115,60 +202,117 @@ set(BREAKPAD_SOURCES_CLIENT_IOS
     breakpad/src/client/mac/handler/ucontext_compat.h
 )
 
+set(BREAKPAD_SOURCES_DUMPSYMS_LINUX
+    breakpad/src/tools/linux/dump_syms/dump_syms.cc
+)
 
+set(BREAKPAD_SOURCES_DUMPSYMS_WINDOWS
+    breakpad/src/tools/windows/dump_syms/dump_syms.cc
+)
+
+set(BREAKPAD_SOURCES_DUMPSYMS_MAC
+    breakpad/src/tools/mac/dump_syms/dump_syms_tool.cc
+)
+
+set(BREAKPAD_SOURCES_MACHODUMP_MAC
+    breakpad/src/tools/mac/dump_syms/macho_dump.cc
+)
+
+add_library(breakpad_common OBJECT)
+add_library(breakpad_common_extra OBJECT EXCLUDE_FROM_ALL)
 add_library(breakpad_client STATIC)
-target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON})
+add_executable(dump_syms)
+
+target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON})
+target_link_libraries(breakpad_client PRIVATE $<BUILD_INTERFACE:breakpad_common>)
+
+target_sources(breakpad_common_extra PRIVATE ${BREAKPAD_SOURCES_COMMON_EXTRA})
+target_link_libraries(breakpad_common_extra PUBLIC breakpad_common)
+
+target_link_libraries(dump_syms PRIVATE breakpad_common breakpad_common_extra)
 
 if(LINUX OR ANDROID)
-    target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON_LINUX} ${BREAKPAD_SOURCES_CLIENT_LINUX})
+    target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_LINUX})
+    target_sources(breakpad_common_extra PRIVATE ${BREAKPAD_SOURCES_COMMON_EXTRA_UNIX} ${BREAKPAD_SOURCES_COMMON_EXTRA_LINUX})
+    target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_CLIENT_LINUX})
+    target_sources(dump_syms PRIVATE ${BREAKPAD_SOURCES_DUMPSYMS_LINUX})
     if(ANDROID)
-        target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON_ANDROID})
-        target_include_directories(breakpad_client PRIVATE breakpad/src/common/android/include)
-    endif(ANDROID)
+        target_sources(breakpad_common PRIVATE ${BREAKPAD_SOURCES_COMMON_ANDROID})
+        target_include_directories(breakpad_common PUBLIC src/common/android/include)
+    endif()
+        if(LINUX)
+            find_package(Threads REQUIRED)
+            target_link_libraries(breakpad_common_extra PRIVATE Threads::Threads)
+        endif()
 
     include(CheckFunctionExists)
     check_function_exists(getcontext HAVE_GETCONTEXT)
     if(HAVE_GETCONTEXT)
-        target_compile_definitions(breakpad_client PRIVATE HAVE_GETCONTEXT)
-    else()
-        target_sources(breakpad_client PRIVATE ${BREAKPAD_SOURCES_COMMON_LINUX_GETCONTEXT})
```

Comment:
why exactly was this changed? I thought this file was only needed as a polyfill if `getcontext` is not available on the system?

Reply:
source can be included in the project unconditionally (it adds `breakpad_getcontext` function which will be stripped by linker) but L251 is an error, it makes `check_function_exists` call useless and breaks build for some libc implementations like musl.

Reply:
I misunderstood `HAVE_GETCONTEXT`. I thought it was a user option but apparently it is the result of `check_function_exists`.
This needs to be reverted.